src/Controller/SentencesController.php

```diff
@@ -828,6 +833,61 @@ public function edit_correctness()
         }
     }
 
+    /**
+     * Mark all sentences of a user as incorrect.
+     *
+     * @param string $userId ID of the user. 
+     * 
+     * @return void
+     */
+    public function mark_unreliable($userId)
+    {   
+        $this->loadModel('Users');
+        $user = $this->Users->getUserById($userId);
+
+        if(empty($user)) {
+            $this->set("userExists", false);
+            $this->redirect(
+                array(
+                    "controller" => "sentences",
+                    "action" => "of_user",
+                    $user['username']
+                )
+            );
+            return;
+        }
+
+        if (CurrentUser::canMarkSentencesOfUser($user)) {
+            $marked = $this->Sentences->markUnreliable($userId);
+            
+            if($marked) {
+                $this->Flash->set(format(
+                    __d('admin', 'Marked all sentences added by {username} as unreliable.'),
+                    ['username' => $user->username]
```

Comment:
Since we want to give the admin feedback and we need the username I think after all it is better not to pass the id but pass the username as you did earlier. For the view and the model it doesn't really matter but here we would need to do an extra database fetch.

Reply:
That makes sense, but that would mean that everything in `markUnreliable` would have to use the username to avoid that extra fetch, instead of using the ID. Is that alright? (else, we'll just be delaying the fetch)  Considering that usernames are unique it shouldn't be a problem.

Reply:
Took a look at the code again. If we pass username to `markUnreliable`, we would still need a query to get the ID, so that we can fetch the user to perform the security check, with the current code structure. (._.)

Reply:
In `markUnreliable` you load the user entity for the check, so you have access to all its properties there.
(For loading a user if you know the username you can use [`Users::getInformationOfUser()`](https://github.com/Tatoeba/tatoeba2/blob/cd1a87eba80a5f19ed25f7dd9803915345b3a7dc/src/Model/Table/UsersTable.php#L238).)

Reply:
@AndiPersti Ah, I didn't notice the method. Fixed it now.